src/Microsoft.ML.Data/Utilities/ColumnCursor.cs

```diff
@@ -20,16 +20,18 @@ public static class ColumnCursorExtensions
         /// </summary>
         /// <typeparam name="T">The type of the values. This must match the actual column type.</typeparam>
         /// <param name="data">The data view to get the column from.</param>
-        /// <param name="env">The current host environment.</param>
-        /// <param name="columnName">The name of the column to extract.</param>
-        public static IEnumerable<T> GetColumn<T>(this IDataView data, IHostEnvironment env, string columnName)
+        /// <param name="column">The column to be extracted.</param>
+        public static IEnumerable<T> GetColumn<T>(this IDataView data, DataViewSchema.Column column)
         {
-            Contracts.CheckValue(env, nameof(env));
-            env.CheckValue(data, nameof(data));
-            env.CheckNonEmpty(columnName, nameof(columnName));
+            Contracts.CheckValue(data, nameof(data));
+            Contracts.CheckNonEmpty(column.Name, nameof(column));
 
-            if (!data.Schema.TryGetColumnIndex(columnName, out int col))
-                throw env.ExceptSchemaMismatch(nameof(columnName), "input", columnName);
+            if (!data.Schema.TryGetColumnIndex(column.Name, out int colIndex))
```

Comment:
>if (!data.Schema.TryGetColumnIndex(column.Name, out int colIndex) [](start = 12, length = 65)

With this implementation I don't see any reason to have `DataViewSchema.Column column` instead of just `string columnName`.

So imagine I'm doing following:
I have column `A` of type string. I modify that column to `vector<float>` with same name `A`.
Then I apply ToKey to `A`, but preserve name.

In old code, I can say: give me `A` and i would get key A.
with your code, I can inspect `IDataview` and I can find 3 `A` columns, two are hidden and one is not.
I pick `A` which is `vector<float>` and pass to your function.

I would get exception, right?
What's the point of this new function, if it does same thing, but in more obfuscated way by forcing me to get `DataViewSchema.Column` first instead of just name?

Reply:
How about this?
```csharp
            var colIndex = column.Index;
            var colType = column.Type;
            var colName = column.Name;

            // Use column index as the principle address of the specified input column and check if that address in data contains
            // the column indicated.
            if (data.Schema[colIndex].Name != colName || data.Schema[colIndex].Type != colType)
                throw Contracts.ExceptParam(nameof(column), string.Format("column with name {0}, type {1}, and index {2} cannot be found in {3}",
                    colName, colType, colIndex, nameof(data)));
```
We use input `column.Index` to retrieve information from `data`.

Reply:
Looks good.

---
In reply to: [260491233](https://github.com/dotnet/machinelearning/pull/2738#discussion_r260491233) [](ancestors = 260491233)